cmd/tenant.go

```diff
@@ -136,6 +136,9 @@ func tenantApply() error {
 		if t.Payload.Tenant.Meta != nil {
 			params := tenant.NewUpdateTenantParams()
 			params.SetBody(&models.V1TenantUpdateRequest{Tenant: &tar})
+			//necessary to resolv error 412 Failed Precondition -> set Version always to last Version from master DB
+			//UpdateTenant setVersion +1 to new meta.version
+			params.Body.Tenant.Meta.Version = t.Payload.Tenant.Meta.Version
```

Comment:
We should not update Meta.Version from the client side. This is done on the Server side on a update: 
https://github.com/metal-stack/masterdata-api/blob/master/pkg/datastore/postgres.go#L215

The version is used to detect and prevent optimistic locking.

So please remove this. If a error occurs during update, you have sent a older version of the object than stored in the database.

Reply:
If no version number or a small one is given, processing is terminated. This is particularly uncomfortable without any version number. The server does not automatically set the correct version number.

Reply:
i fyou want to change the description for a tenant like the given file:
>apiversion: v1
description: Demo-Tenant 12 
kind: Tenant
name: Demo
meta:
  id: demo
>
you get the following error from the cli:
`Error: An http error has occurred (status code: 412): rpc error: code = FailedPrecondition desc = optimistic lock error updating tenant with id demo, existing version 30 mismatches entity version 0`
same error exists if you try to check in a file with an older version:
>apiversion: v1
description: Demo-Tenant 12 
kind: Tenant
name: Demo
meta:
  id: demo
  version: 20
>
`Error: An http error has occurred (status code: 412): rpc error: code = FailedPrecondition desc = optimistic lock error updating tenant with id demo, existing version 30 mismatches entity version 20` 
i think its not necessary for the user to know the latest version, the version should be managed from the server. can you please rethink about this issue?

Reply:
The reason for the version check is to prevent the user to override properties of a tenant with invalid values which might have been updated since the current user has.

The version must always be submitted, faking this on the client side will hide these type of errors.

Still needs to be removed.

Reply:
In the first comment you said the server will manage the versions in the second comment you saif the version must always be submitted?

I think it's weird that a user has to figure out a version number for updating the tenant entity. What kind of user experience is this? Isn't this more error-prone than anything? In Kubernetes I also do not have to come up with my own "revision" management for resources to update an entity. :confused:

Reply:
Your right, i have to look at the spec again...

Reply:
Once you remove the comment, and resolve merge conflict i will merge this